### PR TITLE
fix(auth): auto-generate unique usage_id in create_llm() (#1964)

### PR DESCRIPTION
**Pull request by timon0305**
## Summary

When using OpenAISubscriptionAuth.create_llm() multiple times without explicit usage_id, the LLMRegistry raises ValueError because both LLMs get the default usage_id of "default".

Changes:
- create_llm() now auto-generates unique usage_id if not provided (format: chatgpt-subscription-{8_hex_chars})
- Added generate_usage_id() utility function in llm.py for consistent unique ID generation across the codebase
- Added has() and get_or_none() methods to LLMRegistry for safer registry operations
- Exported generate_usage_id from openhands.sdk.llm package
## Checklist

- [ ] If the PR is changing/adding functionality, are there tests to reflect this?
- [ ] If there is an example, have you run the example to make sure that it works?
- [ ] If there are instructions on how to run the code, have you followed the instructions and made sure that it works?
- [ ] If the feature is significant enough to require documentation, is there a PR open on the OpenHands/docs repository with the same branch name?
- [ ] Is the github CI passing?

### CODE REVIEW

**Review by enyst — COMMENTED** [review]
Thank you for this PR, timon! I apologize for the delay, I’ve been thinking about this and I’m not sure it’s the right fix. Does it happen for subscription?

I would love to know if @VascoSch92 has thoughts, because I think maybe we will meet this problem with various profiles switches too…

File: tests/sdk/llm/test_llm.py [hunk 1]
```diff
@@ -993,4 +993,52 @@ def test_llm_respects_allow_short_context_windows_env_var(mock_get_model_info):
...
+
+def test_generate_usage_id_custom_prefix():
+    """Test generate_usage_id with custom prefix."""
+    usage_id = generate_usage_id("my-agent")
+    assert usage_id.startswith("my-agent-")
+    suffix = usage_id.split("-")[-1]
+    assert len(suffix) == 8
+
+
+def test_generate_usage_id_uniqueness():
```
Comment (VascoSch92) [inline tests/sdk/llm/test_llm.py:1018]: This test is not deterministic. 

I don't know if we want to include that in the test suite.

File: tests/sdk/llm/test_llm.py [hunk 2]
```diff
@@ -993,4 +993,52 @@ def test_llm_respects_allow_short_context_windows_env_var(mock_get_model_info):
...
+    # Should be valid hex
+    int(suffix, 16)
+
+
+def test_generate_usage_id_custom_prefix():
+    """Test generate_usage_id with custom prefix."""
+    usage_id = generate_usage_id("my-agent")
+    assert usage_id.startswith("my-agent-")
+    suffix = usage_id.split("-")[-1]
+    assert len(suffix) == 8
```
Comment (VascoSch92) [inline tests/sdk/llm/test_llm.py:1015]: nit: put the two tests together using `@pytest.mark.parametrize` and then you could also check for some corner case...

File: tests/sdk/llm/test_llm.py [hunk 3]
```diff
@@ -993,4 +993,52 @@ def test_llm_respects_allow_short_context_windows_env_var(mock_get_model_info):
         assert llm.max_input_tokens == 2048
 
 
+# generate_usage_id Tests
```
Comment (VascoSch92) [inline tests/sdk/llm/test_llm.py:996]:
```suggestion
```

File: tests/sdk/llm/test_llm.py [hunk 4]
```diff
@@ -993,4 +993,52 @@ def test_llm_respects_allow_short_context_windows_env_var(mock_get_model_info):
...
+
+def test_generate_usage_id_default_prefix():
+    """Test generate_usage_id with default prefix."""
+    usage_id = generate_usage_id()
+    assert usage_id.startswith("llm-")
+    # Should have 8 hex characters after prefix
+    suffix = usage_id.split("-", 1)[1]
+    assert len(suffix) == 8
+    # Should be valid hex
+    int(suffix, 16)
```
Comment (VascoSch92) [inline tests/sdk/llm/test_llm.py:1007]:
```suggestion
    assert isinstance(int(suffix, 16), int)
```

**Review by VascoSch92 — COMMENTED** [review]
For instance, I don't think we've encountered a situation where this specific problem occurs.

Do we have any documented cases where this caused an error? I’d like to understand what kind of fix is needed—or if one is even necessary.

Aside from that, I’ve left a couple of comments.
